packages/vm/lib/runBlockchain.ts

```diff
@@ -5,13 +5,13 @@ import VM from './index'
 /**
  * @ignore
  */
-export default async function runBlockchain(this: VM, blockchain?: Blockchain, maxBlocks?: number) {
```

Comment:
I would argue that return value can be changed this way here (remaining non-breaking), since no one will have done something with the `void` return value before and just used `await vm.runBlockchain()`. So the updated return value is more of an optional extra thing now which now can be used if makes sense.

Reply:
great, sounds good, could the same reasoning be used for the return type of `iterator`? I think I would find `void | number` confusing.

Reply:
Hmm, not for the interface I guess, since this would mean that if someone implements that interface it would break their implementation. 😕  I've added this (remove `void` return type part) to #1024 though, and the `iterator` implementation itself has no void return type.

Reply:
When thinking about it I'm afraid that it's rather the other way around and we need to also return `Promise<void | number>` in the VM for the non-breaking release. Otherwise if someone keeps their own `Blockchain` implementation with the `void` return and uses this for the `VM.runBlockchain()` method, this will break on `TypeScript` since the method is expecting a chained return value of `number` for the output.